src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs

```diff
@@ -62,29 +79,19 @@ public Task<AckStatus> CreateMultiAck(out int id, TimeSpan? ackTimeout = default
             {
                 return Task.FromResult(AckStatus.Ok);
             }
-            var info = (IAckInfo<AckStatus>)_acks.GetOrAdd(id, _ => new MultiAckInfo(ackTimeout ?? _defaultAckTimeout));
-            if (info is SingleAckInfo)
+            var info = (IAckInfo<AckStatus>)_acks.GetOrAdd(id, _ => new MultiAckWithStatusInfo(ackTimeout ?? _defaultAckTimeout));
+            if (info is SingleAckInfo<AckStatus>)
             {
                 throw new InvalidOperationException();
             }
             return info.Task;
         }
 
-        public void TriggerAck(int id, AckStatus status = AckStatus.Ok)
+        public void TriggerAck(int id, AckStatus status = AckStatus.Ok, ReadOnlySequence<byte>? payload = default)
         {
-            if (_acks.TryGetValue(id, out var info))
+            if (_acks.TryGetValue(id, out var info) && info.Ack(status, payload))
             {
-                switch (info)
-                {
-                    case IAckInfo<AckStatus> ackInfo:
-                        if (ackInfo.Ack(status))
-                        {
-                            _acks.TryRemove(id, out _);
-                        }
-                        break;
-                    default:
-                        throw new InvalidCastException($"Expected: IAckInfo<{typeof(IAckInfo<AckStatus>).Name}>, actual type: {info.GetType().Name}");
-                }
+                _acks.TryRemove(id, out _);
```

Comment:
changes the original logic of type check?

Reply:
Yes, as we've move the `Ack(...)` method up from `IAckInfo<T>` to `IAckInfo`, there is no need for type check.